cmd/lotus-chainwatch/syncer/sync.go

```diff
@@ -264,6 +277,24 @@ func (s *Syncer) syncedBlocks(timestamp time.Time) (map[cid.Cid]struct{}, error)
 	return out, nil
 }
 
+func (s *Syncer) storeCirculatingSupply(ctx context.Context, tipset *types.TipSet) error {
+	supply, err := s.node.StateCirculatingSupply(ctx, tipset.Key())
+	if err != nil {
+		return err
+	}
+
+	ceInsert := `insert into chain_economics (parent_state_root, circulating_fil) values ('%s', '%s');`
```

Comment:
any reason to not construct the complete insert statement here?

Reply:
To make the data being added with the `db.Exec` call more obvious. (Pretend this is a prepare. :P )